Frameworks/CoreText/CTFramesetter.mm

```diff
@@ -23,30 +23,31 @@
 
 using namespace std;
 
-@implementation _CTFrameSetter : NSObject
+@implementation _CTFramesetter : NSObject
 - (void)dealloc {
+    _typesetter = nil;
     [super dealloc];
 }
 @end
 
-static _CTFrame* _createFrame(_CTFrameSetter* frameSetter, CGRect frameSize, CGSize* sizeOut, bool createFrame) {
+static _CTFrame* _createFrame(_CTFramesetter* framesetter, CGRect frameSize, CGSize* sizeOut, bool createFrame) {
     _CTFrame* ret = nil;
 
     if (createFrame) {
         ret = [_CTFrame new];
-        ret->_frameSetter = [frameSetter retain];
```

Comment:
>[frameSetter retain]; [](start = 28, length = 21)

Why have you removed the retain here? framesetter object is being handed over to CTFrame and it should hold a reference to it right?